lib/engine/assignable.rb

```diff
@@ -6,16 +6,23 @@ def assignments
       @assignments ||= {}
     end
 
+    def assignment_stack_groups
+      @assignment_stack_groups ||= {}
+    end
+
     def assigned?(key)
       assignments.key?(key)
     end
 
-    def assign!(key, value = true)
+    def assign!(key, value = true, stack_group: 'NONE')
```

Comment:
If this is the approach taken* then the `stack_group` parameter should be a label, not a string:
```suggestion
    def assign!(key, value = true, stack_group: :none)
```
[* I'm still not keen on this. The presentation of the assignments is an interface decision, so I believe the code for managing stacks of assignments should be done in the game view, not the game logic for assigning stacks to hexes.]

Reply:
I will try to rewrite on game level instead.
How should I then add the number should I have a unique icon for each number or are there a better solution?

Reply:
Do you mean the number of assignments shown in the circle? I like the current approach of drawing a circle and having the number as a text node in the SVG.

Reply:
Re-read your comment, the suggestion is rather to add it to game view not necesserly to the game?
Your main objection is that this proposal mixes the game logic for assignments with the graphical presentation, will see if I can find a better way to do this.

Reply:
I was thinking that you could leave the current code for `assign!` unchanged, and then modify `View::Game::Part::Assignments` to automatically stack assignments. In that class you could make a call to something like `@game.stack_assignments(tile, assignments)` and that method would be able to determine if there are multiple assignments of the same type and decide if they should be stacked. The default implementation should not change how existing games render this, so maybe don't stack assignments if there are less than six of the same type. The stacking limit could be defined as a constant, so this could be easily overridden in the 18Uruguay code.

Reply:
Let us see if I understood you correctly. Have updated with a new proposal that I think matches your suggestion